### PR TITLE
Add option to remove leading zeroes from ghost tags

### PR DESCRIPTION
**Pull request by riidefi**
Fully addresses #58:
- Added `SP_TA_RULE_GHOST_TAG_CONTENT_TIME_NOLEADING` to describe a ghost finish time without leading zeroes.
- Added a "Shortened Time" option to the UI to select `SP_TA_RULE_GHOST_TAG_CONTENT_TIME_NOLEADING`.

Screenshots:
image
image

Admittedly "Shortened Time" isn't the best name, but we're very constrained as to what can fit in the UI slot.

### CODE REVIEW

File: assets/message/Menu_F.bmg.json5 [hunk 1]
```diff
@@ -5623,6 +5623,10 @@
         "font": "regular",
         "string": "Confirmer ces règles.",
     },
+    "0x302f": {
```
Comment (stblr) [inline assets/message/Menu_F.bmg.json5:5626]: Can you make this 0x3036, 0x302f is already used in Race.bmg and I'd like to keep message ids unique.

File: assets/message/Menu_E.bmg.json5 [hunk 2]
```diff
@@ -5619,9 +5619,9 @@
         "font": "regular",
         "string": "Make all ghosts solid.",
     },
-    "0x302e": {
+    "0x302f": {
```
Comment (stblr) [inline assets/message/Menu_E.bmg.json5:5622]: Looks like something happened to this one.

File: payload/game/ui/ctrl/CtrlRaceNameBalloon.c [hunk 3]
```diff
@@ -52,7 +52,7 @@ static void CtrlRaceNameBalloon_refreshTextName(CtrlRaceNameBalloon *this, u32 p
     LayoutUIControl_setMessage(this, "chara_name", 0x251d, &info);
 }
 
-static void CtrlRaceNameBalloon_refreshTextTime(CtrlRaceNameBalloon *this, u32 playerId) {
+static void CtrlRaceNameBalloon_refreshTextTime(CtrlRaceNameBalloon *this, u32 playerId, bool leading_zeroes) {
```
Comment (stblr) [inline payload/game/ui/ctrl/CtrlRaceNameBalloon.c:55]: We use camelCase for parameters.

**Review by stblr — CHANGES_REQUESTED** [review]
Otherwise LGTM, nice work.

**Comment by riidefi** [comment]
Thanks 😃 I've pushed a fix to everything you've brought up.

**Review by stblr — APPROVED** [review]

**Comment by stblr** [comment]
Can you rebase (and optionally squash)?

**Pull request merged**